### PR TITLE
Add client-side pagination

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

https://github.com/tektoncd/dashboard/issues/2324

Add client-side pagination to all list pages via the `ListPageLayout`
container. Update the `children` prop to expect a function so we can
use the render props pattern to allow the `ListPageLayout` component
to handle all of the pagination logic required.

Consuming pages pass their raw resources to `ListPageLayout` and it
passes back the correct slice for the current pagination settings
which the consumer then formats / renders as needed.

Although we're still loading and storing the full list of resources
from the API this does have significant performance benefits in the
client, especially with larger number of resources. Notably, this
addresses freezing reported by a number of users on the PipelineRuns
and TaskRuns pages with busy clusters.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/main/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: src/containers/ListPageLayout/ListPageLayout.js [hunk 1]
```diff
@@ -72,6 +88,11 @@ export const ListPageLayout = ({
     setPath(newURL);
   }
 
+  const resourcesForCurrentPage = resources.slice(
+    (page - 1) * pageSize,
+    (page - 1) * pageSize + pageSize
```
Comment (briangleeson) [inline src/containers/ListPageLayout/ListPageLayout.js:93]: Could simplify to `page * pageSize` for the second value?

Comment (AlanGreene) [reply src/containers/ListPageLayout/ListPageLayout.js:93]: absolutely 🥇

Comment (AlanGreene) [reply src/containers/ListPageLayout/ListPageLayout.js]: Done

**Review by briangleeson — APPROVED** [review]
/lgtm

**Pull request merged**